src/main/java/VASSAL/launch/ModuleManager.java

```diff
@@ -411,7 +411,14 @@ public void run() {
             ErrorDialog.bug(e);
           }
           finally {
-            IOUtils.closeQuietly(clientSocket);
```

Comment:
Is there a reason why we can't do `try (Socket clientSocket = serverSocket.accept()) {` at the head of the loop body and avoid the `finally` block completely?

Reply:
Honestly, I am afraid of changing it that much due to these two lines in the middle of the while loop and the lack of unit tests:

```
clientSocket.close();
if (message == null || clientSocket.isClosed()) continue;
```

I have no idea what is going on there, why there's a need to check if the socket is closed right after calling `close()`, and when (if ever) the code after the `continue` is supposed to be executed, given normal circumstances where no exceptions are thrown.

Reply:
I added `clientSocket.isClosed()` there in `226c2deb`, noting:
```
Fixed Bug 2489325: SocketException in ModuleManager$SocketListener.run()
Check whether client is listening before trying to write reply.
```
I believe the problem that addressed was that the other end would sometimes hang up before a response could be written. I'm 50-50 on whether writing to the socket after this is a bug or intentional; Java's sockets implementation behaves unexpectedly in my experience, so I could certainly imagine `isClosed()` returning false here and the other end to still be waiting for a response.

Probably best to leave that alone for now unless you want to dig into testing it (noting that it's devilishly hard to test sockets code).

Reply:
"given normal circumstances where no exceptions are thrown" ---> That's the rub with sockets in Java. You do get exceptions in situations which are totally normal, such as the other end hanging up.

Reply:
> Probably best to leave that alone for now unless you want to dig into testing it (noting that it's devilishly hard to test sockets code).

I think so too, for now this is a good first step, it removes the deprecated `closeQuietly()`, makes the second `close` inside the finally block more verbose, and maybe even highlights a previously unknown/unnoticed problem, while keeping the exact same functionality it had before.